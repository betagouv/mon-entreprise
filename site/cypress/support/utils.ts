export const fr = Cypress.env('language') === 'fr'
export const baseUrl = Cypress.config('baseUrl')
