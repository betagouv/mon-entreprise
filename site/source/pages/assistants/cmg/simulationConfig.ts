import { SimulationConfig } from '@/domaine/SimulationConfig'

export const configCMG: SimulationConfig = {}
