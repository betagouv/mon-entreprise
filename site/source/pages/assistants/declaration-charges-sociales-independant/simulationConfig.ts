import { SimulationConfig } from '@/domaine/SimulationConfig'

export const config: SimulationConfig = {
	'objectifs exclusifs': [
		'déclaration charge sociales . résultat . cotisations obligatoires',
		'déclaration charge sociales . résultat . total charges sociales déductible',
		'déclaration charge sociales . résultat . revenu net fiscal',
		'déclaration charge sociales . résultat . assiette sociale',
	],
	situation: {
		'déclaration charge sociales': 'oui',
		'dirigeant . régime social': "'indépendant'",
		date: '01/01/2024',
		'dirigeant . indépendant . PL . CIPAV': 'non',
		'entreprise . catégorie juridique': "''",
		impôt: 'non',
	},
	'unité par défaut': '€',
}
