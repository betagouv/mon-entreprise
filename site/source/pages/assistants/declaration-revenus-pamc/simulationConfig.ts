import { SimulationConfig } from '@/domaine/SimulationConfig'

export const configDéclarationRevenusPAMC: SimulationConfig = {
	situation: {
		'déclaration revenus PAMC': 'oui',
	},
	'unité par défaut': '€',
}
