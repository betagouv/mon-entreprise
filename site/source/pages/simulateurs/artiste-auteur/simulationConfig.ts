import { SimulationConfig } from '../_configs/types'

export const configArtisteAuteur: SimulationConfig = {
	objectifs: [
		'artiste-auteur . cotisations',
		'artiste-auteur . revenus',
		'artiste-auteur . cotisations . IRCEC',
	],
	situation: {
		'artiste-auteur': 'oui',
		dirigeant: {
			'applicable si': 'non',
		},
	},
	'unité par défaut': '€/an',
	questions: {
		liste: [
			'artiste-auteur . revenus . BNC . micro-bnc',
			'artiste-auteur . cotisations . option surcotisation',
			'artiste-auteur . cotisations . IRCEC . cotisation RAAP . taux réduit',
			'artiste-auteur . cotisations . IRCEC . profession',
		],
	},
}
