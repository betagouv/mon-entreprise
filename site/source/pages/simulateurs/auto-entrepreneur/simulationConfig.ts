import { SimulationConfig } from '../_configs/types'

export const configAutoEntrepreneur: SimulationConfig = {
	'objectifs exclusifs': [
		"dirigeant . auto-entrepreneur . chiffre d'affaires",
		'dirigeant . auto-entrepreneur . revenu net',
		'dirigeant . auto-entrepreneur . revenu net . après impôt',
	],
	objectifs: [
		'dirigeant . auto-entrepreneur . cotisations et contributions',
		'dirigeant . rémunération . impôt',
	],
	questions: {
		"à l'affiche": [
			{
				label: "Type d'activité",
				dottedName: 'entreprise . activité . nature',
			},
			{
				label: 'Versement libératoire',
				dottedName:
					'dirigeant . auto-entrepreneur . impôt . versement libératoire',
			},
			{
				label: 'Impôt sur le revenu',
				dottedName: 'impôt . méthode de calcul',
			},
			{
				label: 'ACRE',
				dottedName: 'dirigeant . exonérations . ACRE',
			},
		],
		liste: [
			'entreprise',
			'dirigeant',
			'impôt',
			'établissement',
			'situation personnelle',
		],
		'non prioritaires': ['établissement . commune'],
		'liste noire': [
			'entreprise . activités',
			'entreprise . activités . revenus mixtes',
			'entreprise . charges',
			"entreprise . chiffre d'affaires",
		],
	},
	'unité par défaut': '€/an',
	situation: {
		salarié: 'non',
		'entreprise . catégorie juridique': "'EI'",
		'entreprise . catégorie juridique . EI . auto-entrepreneur': 'oui',
	},
}
