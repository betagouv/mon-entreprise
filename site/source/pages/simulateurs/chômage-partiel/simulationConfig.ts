import { SimulationConfig } from '../_configs/types'

export const configChômagePartiel: SimulationConfig = {
	objectifs: [
		'salarié . contrat . salaire brut',
		'salarié . rémunération . net . à payer avant impôt',
		'salarié . activité partielle . net habituel',
		'salarié . activité partielle . total employeur habituel',
		'salarié . coût total employeur',
	],
	questions: {
		liste: [
			'salarié . activité partielle',
			'salarié . temps de travail',
			'établissement . commune',
		],
	},
	'unité par défaut': '€/mois',
	situation: {
		dirigeant: 'non',
		'salarié . activité partielle': 'oui',
		'salarié . cotisations . prévoyances': {
			'applicable si': 'non',
		},
	},
}
