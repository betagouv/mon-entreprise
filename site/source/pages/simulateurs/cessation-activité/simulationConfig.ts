import { SimulationConfig } from '@/pages/simulateurs/_configs/types'
import { configIndépendant } from '@/pages/simulateurs/indépendant/simulationConfig'

export const configCessationActivité: SimulationConfig = {
	...configIndépendant,
	'unité par défaut': '€/an',
}
