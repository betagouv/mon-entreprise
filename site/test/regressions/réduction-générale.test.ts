import { it } from 'vitest'

import { configRéductionGénérale } from '@/pages/simulateurs/reduction-generale/simulationConfig'

import réductionGénéraleSituation from './réduction-générale.yaml'
import { runSimulations } from './utils'

it('calculate simulations-dividendes', () => {
	runSimulations(
		réductionGénéraleSituation,
		[
			...(configRéductionGénérale['objectifs exclusifs'] ?? []),
			...(configRéductionGénérale.objectifs ?? []),
			'salarié . cotisations . exonérations . réduction générale . part retraite',
			'salarié . cotisations . exonérations . réduction générale . part Urssaf',
			'salarié . cotisations . exonérations . réduction générale . part Urssaf . part chômage',
		],
		configRéductionGénérale.situation
	)
})
