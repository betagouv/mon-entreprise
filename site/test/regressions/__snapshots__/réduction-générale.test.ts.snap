// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`calculate simulations-dividendes > DFS 1`] = `
"salarié . cotisations . assiette: 1802
salarié . cotisations . exonérations . réduction générale: 523
salarié . cotisations . exonérations . réduction générale . imputation chômage: 73
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 108
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 415"
`;

exports[`calculate simulations-dividendes > DFS 2`] = `
"salarié . cotisations . assiette: 1802
salarié . cotisations . exonérations . réduction générale: 523
salarié . cotisations . exonérations . réduction générale . imputation chômage: 73
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 108
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 415"
`;

exports[`calculate simulations-dividendes > DFS 3`] = `
"salarié . cotisations . assiette: 1802
salarié . cotisations . exonérations . réduction générale: 575
salarié . cotisations . exonérations . réduction générale . imputation chômage: 73
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 108
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 467"
`;

exports[`calculate simulations-dividendes > DFS 4`] = `
"salarié . cotisations . assiette: 1802
salarié . cotisations . exonérations . réduction générale: 575
salarié . cotisations . exonérations . réduction générale . imputation chômage: 73
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 108
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 467"
`;

exports[`calculate simulations-dividendes > DFS 5`] = `
"salarié . cotisations . assiette: 1802
salarié . cotisations . exonérations . réduction générale: 575
salarié . cotisations . exonérations . réduction générale . imputation chômage: 73
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 108
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 467"
`;

exports[`calculate simulations-dividendes > JEI 1`] = `
"salarié . cotisations . assiette: 1900
salarié . cotisations . exonérations . réduction générale: null
salarié . cotisations . exonérations . réduction générale . imputation chômage: null
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: null
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: null"
`;

exports[`calculate simulations-dividendes > effectif 1`] = `
"salarié . cotisations . assiette: 1900
salarié . cotisations . exonérations . réduction générale: 523
salarié . cotisations . exonérations . réduction générale . imputation chômage: 66
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 98
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 425"
`;

exports[`calculate simulations-dividendes > effectif 2`] = `
"salarié . cotisations . assiette: 1900
salarié . cotisations . exonérations . réduction générale: 530
salarié . cotisations . exonérations . réduction générale . imputation chômage: 66
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 98
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 431"
`;

exports[`calculate simulations-dividendes > heures complémentaires 1`] = `
"salarié . cotisations . assiette: 1900
salarié . cotisations . exonérations . réduction générale: 483
salarié . cotisations . exonérations . réduction générale . imputation chômage: 61
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 91
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 392"
`;

exports[`calculate simulations-dividendes > heures supplémentaires 1`] = `
"salarié . cotisations . assiette: 1900
salarié . cotisations . exonérations . réduction générale: 607
salarié . cotisations . exonérations . réduction générale . imputation chômage: 77
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 114
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 493"
`;

exports[`calculate simulations-dividendes > salaire 1`] = `
"salarié . cotisations . assiette: 100
salarié . cotisations . exonérations . réduction générale: 32
salarié . cotisations . exonérations . réduction générale . imputation chômage: 4
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 6
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 26"
`;

exports[`calculate simulations-dividendes > salaire 2`] = `
"salarié . cotisations . assiette: 250
salarié . cotisations . exonérations . réduction générale: 80
salarié . cotisations . exonérations . réduction générale . imputation chômage: 10
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 15
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 65"
`;

exports[`calculate simulations-dividendes > salaire 3`] = `
"salarié . cotisations . assiette: 500
salarié . cotisations . exonérations . réduction générale: 160
salarié . cotisations . exonérations . réduction générale . imputation chômage: 20
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 30
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 130"
`;

exports[`calculate simulations-dividendes > salaire 4`] = `
"salarié . cotisations . assiette: 750
salarié . cotisations . exonérations . réduction générale: 240
salarié . cotisations . exonérations . réduction générale . imputation chômage: 30
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 45
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 194"
`;

exports[`calculate simulations-dividendes > salaire 5`] = `
"salarié . cotisations . assiette: 1000
salarié . cotisations . exonérations . réduction générale: 319
salarié . cotisations . exonérations . réduction générale . imputation chômage: 40
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 60
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 259"
`;

exports[`calculate simulations-dividendes > salaire 6`] = `
"salarié . cotisations . assiette: 1250
salarié . cotisations . exonérations . réduction générale: 399
salarié . cotisations . exonérations . réduction générale . imputation chômage: 51
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 75
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 324"
`;

exports[`calculate simulations-dividendes > salaire 7`] = `
"salarié . cotisations . assiette: 1500
salarié . cotisations . exonérations . réduction générale: 479
salarié . cotisations . exonérations . réduction générale . imputation chômage: 61
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 90
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 389"
`;

exports[`calculate simulations-dividendes > salaire 8`] = `
"salarié . cotisations . assiette: 2000
salarié . cotisations . exonérations . réduction générale: 470
salarié . cotisations . exonérations . réduction générale . imputation chômage: 60
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 88
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 382"
`;

exports[`calculate simulations-dividendes > salaire 9`] = `
"salarié . cotisations . assiette: 2500
salarié . cotisations . exonérations . réduction générale: 204
salarié . cotisations . exonérations . réduction générale . imputation chômage: 26
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 38
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 165"
`;

exports[`calculate simulations-dividendes > salaire 10`] = `
"salarié . cotisations . assiette: 3000
salarié . cotisations . exonérations . réduction générale: 0
salarié . cotisations . exonérations . réduction générale . imputation chômage: 0
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 0
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 0"
`;

exports[`calculate simulations-dividendes > stage 1`] = `
"salarié . cotisations . assiette: 1900
salarié . cotisations . exonérations . réduction générale: 523
salarié . cotisations . exonérations . réduction générale . imputation chômage: 66
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 98
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 425"
`;

exports[`calculate simulations-dividendes > temps partiel 1`] = `
"salarié . cotisations . assiette: 1900
salarié . cotisations . exonérations . réduction générale: 392
salarié . cotisations . exonérations . réduction générale . imputation chômage: 50
salarié . cotisations . exonérations . réduction générale . imputation retraite complémentaire: 74
salarié . cotisations . exonérations . réduction générale . imputation sécurité sociale: 318"
`;
